src/host/consoleInformation.cpp

```diff
@@ -44,42 +46,41 @@ CONSOLE_INFORMATION::CONSOLE_INFORMATION() :
 {
     ZeroMemory((void*)&CPInfo, sizeof(CPInfo));
     ZeroMemory((void*)&OutputCPInfo, sizeof(OutputCPInfo));
-    InitializeCriticalSection(&_csConsoleLock);
 }
 
-CONSOLE_INFORMATION::~CONSOLE_INFORMATION()
-{
-    DeleteCriticalSection(&_csConsoleLock);
-}
+static thread_local ULONG recursionCount = 0;
+static til::ticket_lock lock;

```

Comment:
Ah btw: You can't have thread locals as class members. They only work like that as statics.